engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -386,6 +472,38 @@ public PhysicalPlan(LogicalPlan dag, PlanContext ctx) {
     this.undeployOpers.clear();
   }
 
+  public void setAntiAffinityForContainers(LogicalPlan dag, Collection<AffinityRule> affinityRules, Map<PTOperator, PTContainer> operatorContainerMap)
```

Comment:
Both are possible ways of doing it. This is more consistent with how node locality is handled i.e. PTOperator contains list of operators that should be co-allocated. Advantage of having the anti-affinity list of containers in each container is it makes it easy to assign host to container requests based on anti-affinity list.